### PR TITLE
feat: Add Resources as optional to ModuleTemplate

### PR DESCRIPTION
**Pull request by c-pius**
**Description**

Changes proposed in this pull request:

- adds new optional `Resources` field to ModuleTemplate
- adds documentation for this list
- does NOT add tests since KLM does not make use of this field itself

**Related issue(s)**

- resolves https://github.com/kyma-project/lifecycle-manager/issues/1842

### CODE REVIEW

**Comment by c-pius** [comment]
Docs to be added once https://github.com/kyma-project/lifecycle-manager/pull/1905 is merged:

> ### **.spec.resources**
> 
> The `resources` field is a list of additional resources of the module that can be fetched. As of now, the primary expected use case is for module teams to add a link to the raw manifest of the module.
>

File: api/v1beta2/moduletemplate_types.go [hunk 1]
```diff
@@ -148,6 +154,14 @@ type ModuleTemplateList struct {
 	Items              []ModuleTemplate `json:"items"`
 }
 
+type Resource struct {
+	// Name is the name of the resource.
+	Name string `json:"name"`
```
Comment (nesmabadr) [inline api/v1beta2/moduletemplate_types.go:159]: I was wondering if we want to somehow validate that if the resources array is not empty, then there should be an entry which has the name of "rawManifest"? I am not aware of any built in Kubebuilder markers that can validate this, but in general we do need to have this validation or?

Comment (c-pius) [reply api/v1beta2/moduletemplate_types.go:159]: I think there would be "min items" (at least in JSON schema). But I was actually thinking about the same... The rawManifest is only necessary if the module supports manual installation which is not the case for all. But what we could do is to raise a warning in modulectl that "when missing rawManfiest in resources, the module template does not support manual installation".

Comment (c-pius) [reply api/v1beta2/moduletemplate_types.go:159]: Also added this as a dicussion point to the modulectl related ticket

**Review by nesmabadr — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**